kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -598,4 +603,19 @@ void KeyValueBlockchain::writeSTLinkTransaction(const BlockId block_id, RawBlock
   block_chain_.setAddedBlockId(new_block_id);
 }
 
+std::string KeyValueBlockchain::getPruningStatus() {
+  std::ostringstream oss;
+  std::unordered_map<std::string, std::string> result;
+
+  result.insert(toPair(getName(versioned_num_of_deletes_keys_), versioned_num_of_deletes_keys_.Get().Get()));
+  result.insert(toPair(getName(immutable_num_of_deleted_keys_), immutable_num_of_deleted_keys_.Get().Get()));
+  result.insert(toPair(getName(merkle_num_of_deleted_keys_), merkle_num_of_deleted_keys_.Get().Get()));
+  result.insert(toPair(getName(getGenesisBlockId()), getGenesisBlockId()));
+  result.insert(toPair(getName(getLastReachableBlockId()), getLastReachableBlockId()));
+  result.insert(toPair("isPruningInProgress", bftEngine::ControlStateManager::instance().getPruningProcessStatus()));
```

Comment:
You should make this an `atomic_bool`: https://github.com/vmware/concord-bft/blob/e3661fcd8be966969625e57bb4089fcfe9402642/bftengine/include/bftengine/ControlStateManager.hpp#L79

Reply:
Yes, made it `atomic_bool` for making it thread safe